src/main/java/org/generationcp/middleware/api/germplasm/GermplasmServiceImpl.java

```diff
@@ -1161,6 +1165,63 @@ public long countGermplasmByStudy(final Integer studyDbId) {
 		return this.daoFactory.getGermplasmDao().countGermplasmByStudy(studyDbId);
 	}
 
+	@Override
+	public GermplasmDto getGermplasmDtoById(final Integer gid) {
+		final GermplasmDto germplasmDto = this.daoFactory.getGermplasmDao().getGermplasmDtoByGid(gid);
+		if (germplasmDto != null) {
+			germplasmDto.setNames(this.daoFactory.getNameDao().getGermplasmNamesByGids(Collections.singletonList(gid)));
+			germplasmDto.setGermplasmOrigin(this.daoFactory.getGermplasmStudySourceDAO().getGermplasmOrigin(gid));
+			final List<Progenitor> progenitors = this.daoFactory.getProgenitorDao().getByGID(gid);
+			germplasmDto.setOtherProgenitors(progenitors.stream().map(p -> p.getProgenitorGid()).collect(Collectors.toList()));
+		}
+		return germplasmDto;
+	}
+
+	@Override
+	public ProgenitorsDetailsDto getGermplasmProgenitorDetails(final Integer gid) {
+		final GermplasmDto germplasmDto = this.daoFactory.getGermplasmDao().getGermplasmDtoByGid(gid);
+		if (germplasmDto != null) {
+			final List<Progenitor> progenitors = this.daoFactory.getProgenitorDao().getByGID(gid);
+			germplasmDto.setOtherProgenitors(progenitors.stream().map(p -> p.getProgenitorGid()).collect(Collectors.toList()));
+			final Method method = this.daoFactory.getMethodDAO().getById(germplasmDto.getBreedingMethodId());
+
+			final ProgenitorsDetailsDto progenitorsDetailsDto = new ProgenitorsDetailsDto();
```

Comment:
Create constructor with fields to set

Reply:
that was me... I found that having constructors with a bunch of parameters is hard to read and unuseful when most of them are not mandatory. If have to chose I will prefer a Builder than having what we have now, creating a bunch of constructor and then we end passing null